### PR TITLE
[WIP] Rss tile

### PR DESCRIPTION
**Pull request by nileshgulia1**
#291 Right now parsing is done through `rss-parser`. Looking forward to render a correct UI based on the contents.

### CODE REVIEW

File: src/components/manage/Tiles/Rss/Edit.jsx [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+            className="tile-delete-button"
+          >
+            <Icon name={trashSVG} size="18px" />
+          </Button>
+        )}
+      </div>
+    );
+  }
+}
+const mapStateToProps = state => {
```
Comment (sneridagh) [inline src/components/manage/Tiles/Rss/Edit.jsx]: Not used? See later comment as well.

File: src/components/manage/Tiles/Rss/Edit.jsx [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+      url: this.state.url,
+    });
+    if (this.state.url.match('.rss')){
+    (async () => {
+      const parser = new Parser();
+      const myfeed = await parser.parseURL(this.state.url);
+      this.setState({
+        feed: myfeed,
+      });
+      console.log(myfeed, 'this is of type ' + typeof(this.state.feed.items))
```
Comment (sneridagh) [inline src/components/manage/Tiles/Rss/Edit.jsx]: Ping!

File: src/components/manage/Tiles/Rss/View.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * Property types.
+ * @property {Object} propTypes Property types.
+ * @static
+ */
+View.propTypes = {
+  data: PropTypes.objectOf(PropTypes.any).isRequired,
+};
+
+const mapStateToProps = state => {
```
[the remark this answers is not in the document]
Comment (sneridagh) [reply src/components/manage/Tiles/Rss/View.jsx]: I think you got the Redux store wrong... let's talk about it when you have time.
I'm even having doubts if we should use the store at all here.
I would better make the request by hand on the edit and on the view, based on the stored url tile props, doing a single request on componentDidMount (on the view) and on url set (on edit).

**Comment by sneridagh** [comment]
Fix build, add them to the code too.
Please also, rebase and put the changelog in the right place.

File: src/components/manage/Tiles/Rss/Edit.jsx (hunk 2 above)
Comment (nileshgulia1) [reply src/components/manage/Tiles/Rss/Edit.jsx]: ah. crap! sorry, I used it for testing..

File: src/config/Tiles.jsx [hunk 4]
```diff
@@ -29,6 +30,10 @@ const messagesTiles = defineMessages({
     id: 'video',
     defaultMessage: 'Video',
   },
+  rss: {
```
[the remark this answers is not in the document]
Comment (nileshgulia1) [reply src/config/Tiles.jsx]: Done!

File: src/components/manage/Tiles/Rss/Edit.jsx (hunk 1 above)
Comment (nileshgulia1) [reply src/components/manage/Tiles/Rss/Edit.jsx]: Done!

File: src/components/manage/Tiles/Rss/View.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nileshgulia1) [reply src/components/manage/Tiles/Rss/View.jsx]: Done!

**Comment by sneridagh** [comment]
@nileshgulia1 I tested the tile, and it wasn't working. Didn't you say that it was working? I give it a try, and it turns out that the componentDidMount request was not working (as it uses async/await) I still don't fully get it on the web side but in the docs, it says that that way only works on node environments:
https://www.npmjs.com/package/rss-parser#nodejs

On the web you have to use:
https://www.npmjs.com/package/rss-parser#web

So I changed it and got a nice CORS error... 

@tisto I investigated that, and it seems that it's nearly impossible to overcome that in the client... T_T, so we will need to write a server side way. Unless @robgietema has a better idea... Having that in the server side is not the best from the security point of view... so T_T again...

I'm open to suggestions.

**Comment by nileshgulia1** [comment]
@sneridagh It is working on my side only when i enable CORS through an extension, though some of links are not fetched ( like this one:
``` javascript
{this.state.feed.items.forEach(item => (
                      <div>
                        <Feed.Event>{item.title}</Feed.Event>
                        <Feed.Event>{item.author}</Feed.Event>
                      </div>
                    ))}
```
it throws an `undefined` error. Don't know if its the CORS issue.
I'll pull your changes and test it again asap!
/cc @tisto

**Comment by tisto** [comment]
@nileshgulia1 using a locally installed CORS extension is definitely no way to go. I fully agree with @sneridagh here. We have to find another way.

**Comment by sneridagh** [comment]
We can use an SSR route for it :)

**Comment by nileshgulia1** [comment]
This is a stale PR, would reopen if required.

**Pull request closed without merging**